specification/loadtestservice/LoadTestService.Management/models.tsp

```diff
@@ -0,0 +1,307 @@
+}
+
+@doc("The location of quota resource")
+@subscriptionResource
+model Location {
+  @doc("The location of quota in ARM Normalized format like eastus, southeastasia etc.")
+  @key("location")
+  @path
+  @segment("locations")
+  @visibility("query", "read", "update")
+  name: string;
+}
+
+@doc("Quota bucket details object.")
+@parentResource(Location)
+model QuotaResource is ProxyResource<QuotaResourceProperties> {
+  @doc("The quota name.")
+  @key("quotaBucketName")
+  @path
+  @segment("quotas")
+  @extension("x-ms-parameter-location", "method")
+  name: string;
+}
+
+@doc("LoadTest resource properties.")
+model LoadTestProperties {
+  @maxLength(512)
+  @doc("Description of the resource.")
+  description?: string;
+
+  @visibility("read")
+  @doc("Resource provisioning state.")
+  provisioningState?: ResourceState;
+
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "URI is an abbervation and should be upper case"
+  @visibility("read")
+  @maxLength(2083)
+  @doc("Resource data plane URI.")
+  dataPlaneURI?: string;
+
+  @doc("CMK Encryption property.")
+  encryption?: EncryptionProperties;
+}
+
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "CheckQuotaAvailabilityResponse has a allOf relationship with Resource"
+@doc("Check quota availability response object.")
+model CheckQuotaAvailabilityResponse extends Resource {
+  @doc("Check quota availability response properties.")
+  @extension("x-ms-client-flatten", true)
+  properties?: CheckQuotaAvailabilityResponseProperties;
+}
+
+@doc("Check quota availability response properties.")
+model CheckQuotaAvailabilityResponseProperties {
+  @doc("True/False indicating whether the quota request be granted based on availability.")
+  isAvailable?: boolean;
+
+  @doc("Message indicating additional details to add to quota support request.")
+  availabilityStatus?: string;
+}
+
+@doc("Common properties for resources.")
+model Resource {
+  @doc("Fully qualified resource ID for the resource. Ex - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}")
+  @visibility("read")
+  id?: string;
+
+  @doc("The name of the resource.")
+  @visibility("read")
+  name?: string;
+
+  @doc("The type of the resource. E.g. \"Microsoft.Compute/virtualMachines\" or \"Microsoft.Storage/storageAccounts\"")
+  @visibility("read")
+  type?: string;
+
+  @doc("Azure Resource Manager metadata containing createdBy and modifiedBy information.")
+  @visibility("read")
+  systemData?: SystemData;
+}
+
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "QuotaBucketRequest has a allOf relationship with Resource"
+@doc("Request object of new quota for a quota bucket.")
+model QuotaBucketRequest extends Resource {
+  @extension("x-ms-client-flatten", true)
+  @doc("Request object of new quota for a quota bucket.")
+  properties?: QuotaBucketRequestProperties;
+}
```

Comment:
Is this an arm resource? If so, can we use template to define it. Redefine `Resource` model will cause duplication.

Reply:
It is a request body which extends Resource, we cant use ArmResource model as it doesn't have name parameter in it. The QuotaBucketRequest model is already defined in the GA spec to have an AllOf relation with Resource model. Therefore, resource model is redefined here.

Reply:
It will not have an AllOf relation with relation. Will it cause any issue on SDK breaking change as it was coming in CI checks as one of the breaking change.